tests/unit/src/kakarot/instructions/test_block_information.cairo

```diff
@@ -12,11 +12,77 @@ from starkware.starknet.common.syscalls import get_block_number, get_block_times
 from utils.utils import Helpers
 from kakarot.model import model
 from kakarot.stack import Stack
-from kakarot.constants import Constants
+from kakarot.constants import Constants, blockhash_registry_address
 from kakarot.execution_context import ExecutionContext
 from kakarot.instructions.block_information import BlockInformation
 from tests.unit.helpers.helpers import TestHelpers
 
+// Storage for testing BLOCKHASH
+@storage_var
+func block_number() -> (block_number: Uint256) {
+}
+
+@storage_var
+func blockhash() -> (blockhash: felt) {
+}
+
+// Constructor
+@constructor
+func constructor{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}(block_number_: Uint256, blockhash_: felt, blockhash_registry_address_: felt) {
+    block_number.write(block_number_);
+    blockhash.write(blockhash_);
+    blockhash_registry_address.write(blockhash_registry_address_);
+    return ();
+}
+
+@view
+func test__blockhash_should_push_blockhash_to_stack{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}() {
+    // Given # 1
+    alloc_locals;
+    let (bytecode) = alloc();
+    let stack: model.Stack* = Stack.init();
+
+    let (block_number_: Uint256) = block_number.read();
+
+    let stack: model.Stack* = Stack.push(stack, block_number_);
+    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_stack(0, bytecode, stack);
+
+    // When
+    let result = BlockInformation.exec_blockhash(ctx);
+
+    // Then
+    assert result.gas_used = 20;
+    let len: felt = result.stack.len_16bytes / 2;
+    assert len = 1;
+    let (stack, index0) = Stack.peek(result.stack, 0);
+    let (blockhash_) = blockhash.read();
+    let value = Helpers.to_uint256(val=blockhash_);
+    assert index0 = value;
+
+    // Given # 2
+    let (bytecode) = alloc();
+    let stack: model.Stack* = Stack.init();
+
+    let stack: model.Stack* = Stack.push(stack, Uint256(10, 0));
```

Comment:
What does this 10 represent?
Trying to read the 10th block's hash?

Could be written in well-named variable or comments

Reply:
I wanted to test a block number that is out of range but I agree this is kind of vague, I will update this. Thanks!

Reply:
I think this is a good test case